### ISSUE TITLE
 _reactNativeMdm2.default.isSupported is not a function.

### ISSUE THREAD

**Issue by yuxiaoma** [robinpowered/react-native-mdm#14]
Hi @ajwhite 

I am very glad someone has built this bridge for react native to better integrate with MDM. Thank you!
However, I keep getting this error saying
`TypeError: _reactNativeMdm2.default.isSupported is not a function`
or
`TypeError: _reactNativeMdm2.default.getConfiguration is not a function`

I called these function exactly like instructed in your README.md file.
Is something deprecated or changed?

I am on "react-native-mdm": "^1.0.4"

Thanks

**Comment by ajwhite** [comment]
@yuxiaoma could you double check that you've properly linked the Native Module on the native side? It doesn't look like we describe the setup process, apologies. You'd link it natively like you would normally with a RN module; no extra steps with `react-native-mdm`

**Comment by yuxiaoma** [comment]
yes, that turns out to be the issue. thanks for the quick reply.
Guild for RN linking libraries, in case anyone have the same problem:
https://facebook.github.io/react-native/docs/linking-libraries-ios.html

**Comment by ajwhite** [comment]
> FYI the app halts on device, and quit on similator.

Is this happening on Android or iOS?

> `react-native link react-native-mdm`

Looks like we haven't added support for automated linking. In the meantime, reset your changes and [link manually](https://facebook.github.io/react-native/docs/linking-libraries-ios.html)

**Comment by yuxiaoma** [comment]
This is happening on iOS.

I will try to link manually, but it would great if the package has automated linking support.

**Comment by ajwhite** [comment]
>  it would great if the package has automated linking support.

For sure, let's keep this issue open to track that

**Comment by tomaszmarcula** [comment]
Hi, I have a problem with building the app with react-native-mdm library for iOS.  Below is the error which I get:
node_modules/react-native-mdm/ios/RNMobileDeviceManager.h:8:9: fatal error: module 'AppConfigSettingsFramework' not found

I did manual link of the library to my Xcode project. I added AppConfigSettingsFramework.framework and libRNMobileDeviceManager.a libraries to the "Link Binary With Libraries" section in Build Phases and locally it works, but on the build server I get an error. It looks like RNMobileDeviceManager library is build before AppConfigSettingsFramework. How can I enforce proper build order?

**Comment by DiegoNegri** [comment]
Hi, i have the same problem, i solved it opening the RNNMobileDeviceManager.xcodeproj and setting in 'Build Phases' 'AppConfigSettingsFramework' as Target Dependencies.

**Comment by antoniogoulao** [comment]
Hi all. @DiegoNegri and @ajwhite I was facing the same issue after running `yarn add react-native-mdm@2.0.0` and launching the project. I then ran `cd ios && pod install` and launched the project again in the emulator and it worked.

**Comment by puneetkansal04** [comment]
@ajwhite @DiegoNegri can i get Screen time in the IOS using this library. Please provide me detail for this.
